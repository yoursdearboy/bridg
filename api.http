@base=http://127.0.0.1:8000

# Spaces

### Index spaces

GET {{base}}/spaces

## Sites

### Index sites in the space

GET {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/sites

## Subjects

### Index subjects in the space

GET {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects

### Show a subject in the space

GET {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects/f0519f08-d22f-40ff-8800-cee82c4e925b

### Create a subject in the space

POST {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects
Content-Type: application/json

{
  "status": "candidate",
  "status_date": "2025-03-01T12:00:00",
  "performing_biologic_entity": {
    "administrative_gender_code": "M",
    "birth_date": "2000-03-01",
    "death_date": null,
    "death_date_estimated_indicator": false,
    "death_indicator": false,
    "name": {
      "use": "official", "family": "Test", "given": "Dude"
    }
  },
  "assigned_study_site_protocol_version_relationship": [
    "6a8e6e2b-9537-408f-bd92-a5b83ad2e750"
  ]
}

### Lookup a subject

POST {{base}}/spaces/ce946229-9746-46cd-8dd3-b27a2fbfd48a/subjects/lookup
Content-Type: application/json

{
  "performing_biologic_entity": {
    "name": {
      "use": "official", "family": "Test", "given": "Dude"
    }
  }
}

## Persons

### Show a person

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285

### Names

#### Show a person names

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/names

#### Create a person name

POST {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/names
Content-Type: application/json

{
  "use": "official",
  "family": "Works",
  "given": "Good",
  "prefix": "Jr."
}

### Postal addresses

#### Show a person postal addresses

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/postal_addresses

#### Create a person postal address

POST {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/postal_addresses
Content-Type: application/json

{
  "use": "tmp",
  "street": "Pushkina",
  "building": "Kolotushkina",
  "country": "Lukomorje",
  "municipality": "Great tree",
  "state": "",
  "zip": "111111",
}

### Telecommunication addresses

#### Show a person telecommunication addresses

GET {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/telecommunication_addresses

#### Create a person telecommunication address

POST {{base}}/persons/99132ae7-25dd-43ba-808e-cdb3b29af285/telecommunication_addresses
Content-Type: application/json

{
  "use": "h",
  "shema": "tel",
  "address": "8(800)5553535"
}