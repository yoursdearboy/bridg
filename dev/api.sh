#!/bin/sh

ROOTPATH=$(pwd)

export PYTHONPATH=$ROOTPATH

fastapi dev $ROOTPATH/api/main.py
