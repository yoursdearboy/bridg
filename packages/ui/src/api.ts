import {
  Configuration,
  ObservationResultApi,
  PersonsApi,
  SitesApi,
  DefaultApi as SpacesApi,
  SubjectsApi,
} from "api-ts";

const config = new Configuration({
  basePath: "/api",
});

export const sites = new SitesApi(config);
export const spaces = new SpacesApi(config);
export const subjects = new SubjectsApi(config);
export const persons = new PersonsApi(config);
export const observationResult = new ObservationResultApi(config);

observationResult
  .indexSpacesSpaceIdSubjectsSubjectIdObservationObsIdResultGet({
    spaceId: "ce946229-9746-46cd-8dd3-b27a2fbfd48a",
    subjectId: "1944c046-95b3-4cb4-82e2-c789950e29fc",
    obsId: "8ac78da9-3c23-4770-94e9-ebaee8432b4b",
  })
  .then((x) => {
    console.log(x);
  })
  .catch(console.error);

export default {
  sites,
  spaces,
  subjects,
  persons,
};
