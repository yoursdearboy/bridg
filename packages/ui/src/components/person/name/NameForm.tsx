import {
  Alert,
  Button,
  Group,
  LoadingOverlay,
  Stack,
  TextInput,
} from "@mantine/core";
import { useForm } from "@mantine/form";
import type { UseMutationResult } from "@tanstack/react-query";
import type { EntityName, EntityNameData } from "api-ts";
import { useTranslation } from "react-i18next";

interface NameFormProps {
  initialValues: EntityNameData;
  onClose: () => void;
  mutation: UseMutationResult<EntityName, Error, EntityNameData, unknown>;
}

export const NameForm = ({
  initialValues,
  onClose,
  mutation,
}: NameFormProps) => {
  const { t } = useTranslation();

  const form = useForm<EntityNameData>({
    initialValues: {
      family: initialValues.family || "",
      given: initialValues.given || "",
      middle: initialValues.middle || "",
      patronymic: initialValues.patronymic || "",
      prefix: initialValues.prefix || "",
      suffix: initialValues.suffix || "",
    },
    validate: {
      family: (value) =>
        value
          ? null
          : t("fieldRequiredMessage", { fieldName: t("Name.family") }),
      given: (value) =>
        value
          ? null
          : t("fieldRequiredMessage", { fieldName: t("Name.given") }),
    },
  });

  const handleSubmit = (data: EntityNameData) => mutation.mutate(data);

  return (
    <form onSubmit={form.onSubmit(handleSubmit)}>
      <Stack gap="md" pos="relative">
        <LoadingOverlay visible={mutation.isPending} />
        {mutation.error && (
          <Alert color="red" mb="md">
            {mutation.error.message}
          </Alert>
        )}
        <Group grow>
          <TextInput
            label={t("Name.family")}
            withAsterisk
            {...form.getInputProps("family")}
          />
          <TextInput
            label={t("Name.given")}
            withAsterisk
            {...form.getInputProps("given")}
          />
          <TextInput
            label={t("Name.middle")}
            {...form.getInputProps("middle")}
          />
        </Group>

        <details>
          <summary>{t("extra")}</summary>
          <Group grow>
            <TextInput
              label={t("Name.prefix")}
              {...form.getInputProps("prefix")}
            />
            <TextInput
              label={t("Name.suffix")}
              {...form.getInputProps("suffix")}
            />
          </Group>
          <TextInput
            label={t("Name.patronymic")}
            {...form.getInputProps("patronymic")}
          />
        </details>

        <Group justify="flex-end" mt="md">
          <Button
            variant="outline"
            onClick={onClose}
            disabled={mutation.isPending}
          >
            {t("cancel")}
          </Button>
          <Button type="submit" loading={mutation.isPending}>
            {t("submit")}
          </Button>
        </Group>
      </Stack>
    </form>
  );
};
