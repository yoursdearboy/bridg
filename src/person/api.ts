export const get = (id: number) => {
  return fetch(`/api/persons/${id}`).then((x) => x.json())
}

export default {
  get
}
