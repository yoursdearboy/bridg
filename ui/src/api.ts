import { DefaultApi as SpacesApi, Configuration, SubjectsApi } from "bridg-ts";

const config = new Configuration({
  basePath: "/api",
});

export const spaces = new SpacesApi(config);
export const subjects = new SubjectsApi(config);

export default {
  spaces,
  subjects,
};
