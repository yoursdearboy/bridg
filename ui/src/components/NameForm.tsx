import api from "@/api";
import {
  Button,
  Group,
  TextInput,
  Stack,
  LoadingOverlay,
  Alert,
} from "@mantine/core";
import { useForm } from "@mantine/form";
import { useMutation } from "@tanstack/react-query";
import type { Name } from "bridg-ts";
import { useTranslation } from "react-i18next";

interface NameFormProps {
  personId: string;
  onClose: () => void;
  onSuccess: () => void;
}

export const NameForm = ({ personId, onClose, onSuccess }: NameFormProps) => {
  const { t } = useTranslation();

  const form = useForm<Name>({
    initialValues: {
      use: "",
      family: "",
      given: "",
      middle: "",
      patronymic: "",
      prefix: "",
      suffix: "",
    },
    validate: {
      family: (value) => (value ? null : t("Family name is required")),
      given: (value) => (value ? null : t("Given name is required")),
    },
  });

  const mutation = useMutation({
    mutationFn: (name: Name) =>
      api.persons.createPersonsPersonIdNamesPost({
        personId,
        name,
      }),
    onSuccess: () => {
      onSuccess();
    },
  });

  const handleSubmit = (values: Name) => {
    mutation.mutate(values);
  };

  return (
    <form onSubmit={form.onSubmit(handleSubmit)}>
      <Stack gap="md" pos="relative">
        <LoadingOverlay visible={mutation.isPending} />
        {mutation.isError && (
          <Alert color="red" mb="md">
            {mutation.error.message}
          </Alert>
        )}
        <Group grow>
          <TextInput label={t("Use")} {...form.getInputProps("use")} />
          <TextInput label={t("Prefix")} {...form.getInputProps("prefix")} />
          <TextInput label={t("Suffix")} {...form.getInputProps("suffix")} />
        </Group>

        <TextInput
          label={t("Family name")}
          withAsterisk
          {...form.getInputProps("family")}
        />

        <Group grow>
          <TextInput
            label={t("Given name")}
            withAsterisk
            {...form.getInputProps("given")}
          />
          <TextInput
            label={t("Middle name")}
            {...form.getInputProps("middle")}
          />
        </Group>

        <TextInput
          label={t("Patronymic")}
          {...form.getInputProps("patronymic")}
        />

        <Group justify="flex-end" mt="md">
          <Button
            variant="outline"
            onClick={onClose}
            disabled={mutation.isPending}
          >
            {t("Cancel")}
          </Button>
          <Button type="submit" loading={mutation.isPending}>
            {t("Save")}
          </Button>
        </Group>
      </Stack>
    </form>
  );
};
