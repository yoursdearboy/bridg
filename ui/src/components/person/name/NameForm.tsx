import api from "@/api";
import {
  Button,
  Group,
  TextInput,
  Stack,
  LoadingOverlay,
  Alert,
} from "@mantine/core";
import { useForm } from "@mantine/form";
import { useMutation } from "@tanstack/react-query";
import type { Name } from "bridg-ts";
import { useTranslation } from "react-i18next";

interface NameFormProps {
  personId: string;
  onClose: () => void;
  onSuccess: () => void;
}

export const NameForm = ({ personId, onClose, onSuccess }: NameFormProps) => {
  const { t } = useTranslation();

  const form = useForm<Name>({
    initialValues: {
      use: "",
      family: "",
      given: "",
      middle: "",
      patronymic: "",
      prefix: "",
      suffix: "",
    },
    validate: {
      family: (value) =>
        value
          ? null
          : t("fieldRequiredMessage", { fieldName: t("EntityName.family") }),
      given: (value) =>
        value
          ? null
          : t("fieldRequiredMessage", { fieldName: t("EntityName.given") }),
    },
  });

  const mutation = useMutation({
    mutationFn: (name: Name) =>
      api.persons.createPersonsPersonIdNamesPost({
        personId,
        name,
      }),

    onSuccess,
  });

  const handleSubmit = (values: Name) => {
    mutation.mutate(values);
  };

  return (
    <form onSubmit={form.onSubmit(handleSubmit)}>
      <Stack gap="md" pos="relative">
        <LoadingOverlay visible={mutation.isPending} />
        {mutation.isError && (
          <Alert color="red" mb="md">
            {mutation.error.message}
          </Alert>
        )}
        <Group grow>
          <TextInput
            label={t("EntityName.use")}
            {...form.getInputProps("use")}
          />
          <TextInput
            label={t("EntityName.prefix")}
            {...form.getInputProps("prefix")}
          />
          <TextInput
            label={t("EntityName.suffix")}
            {...form.getInputProps("suffix")}
          />
        </Group>

        <TextInput
          label={t("EntityName.family")}
          withAsterisk
          {...form.getInputProps("family")}
        />

        <Group grow>
          <TextInput
            label={t("EntityName.given")}
            withAsterisk
            {...form.getInputProps("given")}
          />
          <TextInput
            label={t("EntityName.middle")}
            {...form.getInputProps("middle")}
          />
        </Group>

        <TextInput
          label={t("EntityName.patronymic")}
          {...form.getInputProps("patronymic")}
        />

        <Group justify="flex-end" mt="md">
          <Button
            variant="outline"
            onClick={onClose}
            disabled={mutation.isPending}
          >
            {t("cancel")}
          </Button>
          <Button type="submit" loading={mutation.isPending}>
            {t("submit")}
          </Button>
        </Group>
      </Stack>
    </form>
  );
};
