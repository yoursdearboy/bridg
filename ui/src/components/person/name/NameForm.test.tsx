import { render } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";

import api from "@/api";
import { NameForm } from "./NameForm";

describe("NameForm", () => {
  it("matches snapshot", () => {
    vi.spyOn(api.persons, "createPersonsPersonIdNamesPost").mockResolvedValue({
      use: "official",
      family: "James",
      given: "Valya",
      middle: "Gold",
      patronymic: "Gold",
      prefix: "Ms.",
      suffix: "Jr.",
    });

    const { asFragment } = render(
      <NameForm
        personId="2703d0bc-7ed4-497c-91c6-30d86a8eb630"
        onClose={() => {}}
        onSuccess={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
