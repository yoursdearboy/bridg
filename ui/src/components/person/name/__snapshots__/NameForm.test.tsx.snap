// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NameForm > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <form>
    <div
      class="m_6d731127 mantine-Stack-root"
      style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start; position: relative;"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        data-grow="true"
        style="--group-child-width: calc(33.333333333333336% - (var(--mantine-spacing-md) - var(--mantine-spacing-md) / 3)); --group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <div
          class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        >
          <label
            class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
            for="mantine-4miilw5eg"
            id="mantine-4miilw5eg-label"
          >
            use
          </label>
          <div
            class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
            data-variant="default"
          >
            <input
              aria-invalid="false"
              class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
              data-path="use"
              data-variant="default"
              id="mantine-4miilw5eg"
              value=""
            />
          </div>
        </div>
        <div
          class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        >
          <label
            class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
            for="mantine-17gjn16kc"
            id="mantine-17gjn16kc-label"
          >
            Prefix
          </label>
          <div
            class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
            data-variant="default"
          >
            <input
              aria-invalid="false"
              class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
              data-path="prefix"
              data-variant="default"
              id="mantine-17gjn16kc"
              value=""
            />
          </div>
        </div>
        <div
          class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        >
          <label
            class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
            for="mantine-2pbrvloro"
            id="mantine-2pbrvloro-label"
          >
            Suffix
          </label>
          <div
            class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
            data-variant="default"
          >
            <input
              aria-invalid="false"
              class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
              data-path="suffix"
              data-variant="default"
              id="mantine-2pbrvloro"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          data-required="true"
          for="mantine-pfqyzivqd"
          id="mantine-pfqyzivqd-label"
        >
          Family name
          <span
            aria-hidden="true"
            class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
          >
             *
          </span>
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-variant="default"
        >
          <input
            aria-invalid="false"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-path="family"
            data-variant="default"
            id="mantine-pfqyzivqd"
            value=""
          />
        </div>
      </div>
      <div
        class="m_4081bf90 mantine-Group-root"
        data-grow="true"
        style="--group-child-width: calc(50% - (var(--mantine-spacing-md) - var(--mantine-spacing-md) / 2)); --group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <div
          class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        >
          <label
            class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
            data-required="true"
            for="mantine-7lzvq3tar"
            id="mantine-7lzvq3tar-label"
          >
            Given name
            <span
              aria-hidden="true"
              class="m_78a94662 mantine-InputWrapper-required mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
            data-variant="default"
          >
            <input
              aria-invalid="false"
              class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
              data-path="given"
              data-variant="default"
              id="mantine-7lzvq3tar"
              value=""
            />
          </div>
        </div>
        <div
          class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
        >
          <label
            class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
            for="mantine-m75qc1lk3"
            id="mantine-m75qc1lk3-label"
          >
            Middle name
          </label>
          <div
            class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
            data-variant="default"
          >
            <input
              aria-invalid="false"
              class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
              data-path="middle"
              data-variant="default"
              id="mantine-m75qc1lk3"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
      >
        <label
          class="m_8fdc1311 mantine-InputWrapper-label mantine-TextInput-label"
          for="mantine-0imsgk2bo"
          id="mantine-0imsgk2bo-label"
        >
          Patronymic
        </label>
        <div
          class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
          data-variant="default"
        >
          <input
            aria-invalid="false"
            class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
            data-path="patronymic"
            data-variant="default"
            id="mantine-0imsgk2bo"
            value=""
          />
        </div>
      </div>
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap; margin-top: var(--mantine-spacing-md);"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-variant="outline"
          style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-outline-hover); --button-color: var(--mantine-color-blue-outline); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline);"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              Cancel
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          style="--button-color: var(--mantine-color-white);"
          type="submit"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              Submit
            </span>
          </span>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
